### ISSUE TITLE
_t suffix is reserved by POSIX

### ISSUE THREAD

**Issue by rofl0r** [faragon/libsrt#20]
https://stackoverflow.com/questions/231760/what-does-a-type-followed-by-t-underscore-t-represent (see 33 point answer)

affects ss_t and friends

**Comment by faragon** [comment]
Hi rofl0r, thank you very much for the feedback! :-)

I thought it was a small "sin" back in the day, but you're right. Do you have any preference? (e.g. t_ss/t_sv/..., sst/svt/..., ss_s/sv_s/...)

**Comment by rofl0r** [comment]
while i prefered short typedefs in my C beginners days, i came to love the more expressive full name (like `struct string`) lately, even though it's longer to type, since it clearly documents to the reader what we're dealing with. a typedef'd type could be anything from a plain numerical type, to a struct/union, or even an array of something or a pointer (*shudder*).
if you do want to have it typedef'd, then i would suggest something like `srt_string` or `srt_vector`, `srt_map`, etc.
HTH!

**Comment by faragon** [comment]
Thank you, I like your proposal. I'll add it soon :-)

**Comment by faragon** [comment]
I've started this. I'm working for solving the issue while keeping the code readable.

**Comment by faragon** [comment]
Hi @rofl0r, I've added a branch called "issue#20" with the changes for fixing the issue. I kept your idea, but using "srt" instead of "libsrt" as prefix, i.e. ss_t to srt_string, sv_t to srt_vector, etc.

The branch passes the tests, but before merging the changes to master I'll probably do more refinements (e.g. changing smsset.c/h to sset.c/h, and other changes to try to keep the result as coherent as possible). The documentation, because being generated by a script, it requires just to run the script (I'll update the documentation page with the changes after merging the changes to master).
